subprojects/core/src/main/java/org/gradle/api/tasks/testing/TestVerificationException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.testing;
```

Comment:
I am not sure this belongs in the `testing` package. It seems like `org.gradle.api.tasks` seems to be fine, given that `StopActionExecutionException` is also there.

Reply:
I think I can go either way on this.  The intention is not to make this the public facing way that tasks signal verification failures. This is just the easy way to do it for Test tasks right now.

If we moved it into org.gradle.api.tasks, would you also want to rename this exception to something more generic?

Reply:
If this shouldn't be public, then it shouldn't be in a public package but in some internal one. Or is `org.gradle.api.tasks.testing` internal?

If we make this public from the start, I'd call it `VerificationException`.